app/services/issue_searcher.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class IssueSearcher
+  SEARCH_LIMIT = 10
+  PROJECT_RESULTS_LIMIT = 10
+  TICKET_OPEN_STATUS = true
+
+  def call(search_term, scope)
+    issues = if search_term.present?
+               search_by_term(search_term, scope)
+             else
+               filter_closed_issues(scope).limit(SEARCH_LIMIT).order(id: :desc)
+             end
+
+    issues.compact.uniq(&:id)
+  end
+
+  private
+
+  def search_by_term(search_term, scope)
+    found_issues = []
+    found_issues << find_by_id(search_term, scope)
```

Comment:
This adds `nil` sometimes.

Reply:
yes, there is a `compact` call done on the final result set